Sources/Widgets/Representations/ResliceCursorContextRepresentation/index.js

```diff
@@ -89,24 +91,64 @@ function vtkResliceCursorContextRepresentation(publicAPI, model) {
     });
   });
 
+  vtkWidgetRepresentation.connectPipeline(model.pipelines.center);
+  publicAPI.addActor(model.pipelines.center.actor);
+
+  publicAPI.setLineThickness = (lineThickness) => {
+    let scaledLineThickness = lineThickness;
+    if (publicAPI.getScaleInPixels()) {
+      const centerCoords = model.pipelines.center.source.getCenter();
+      scaledLineThickness *= publicAPI.getPixelWorldHeightAtCoord(centerCoords);
+    }
+    model.pipelines.axes[0].line.source.setRadius(scaledLineThickness);
+    model.pipelines.axes[1].line.source.setRadius(scaledLineThickness);
+  };
+
   publicAPI.setSphereRadius = (radius) => {
-    model.pipelines.center.source.setRadius(radius);
-    model.pipelines.axes[0].rotation1.source.setRadius(radius);
-    model.pipelines.axes[0].rotation2.source.setRadius(radius);
-    model.pipelines.axes[1].rotation1.source.setRadius(radius);
-    model.pipelines.axes[1].rotation2.source.setRadius(radius);
+    publicAPI.setSphereRadiusOnSphere(radius, model.pipelines.center.source);
+    publicAPI.setSphereRadiusOnSphere(
+      radius,
+      model.pipelines.axes[0].rotation1.source
+    );
+    publicAPI.setSphereRadiusOnSphere(
+      radius,
+      model.pipelines.axes[0].rotation2.source
+    );
+    publicAPI.setSphereRadiusOnSphere(
+      radius,
+      model.pipelines.axes[1].rotation1.source
+    );
+    publicAPI.setSphereRadiusOnSphere(
+      radius,
+      model.pipelines.axes[1].rotation2.source
+    );
   };
 
-  publicAPI.setSphereRadius(0.05);
+  publicAPI.setSphereRadiusOnSphere = (radius, source) => {
+    let scaledRadius = radius;
+    if (publicAPI.getScaleInPixels()) {
+      const centerCoords = source.getCenter();
+      scaledRadius *= publicAPI.getPixelWorldHeightAtCoord(centerCoords);
+    }
+    source.setRadius(scaledRadius);
+  };
+
+  publicAPI.setSphereRadius(7);
 
   function updateRender(state, axis) {
     const color = state.getColor();
     axis.line.actor.getProperty().setColor(color);
     axis.rotation1.actor.getProperty().setColor(color);
     axis.rotation2.actor.getProperty().setColor(color);
 
-    axis.line.source.setPoint1(state.getPoint1());
-    axis.line.source.setPoint2(state.getPoint2());
+    const vector = [0, 0, 0];
+    vtkMath.subtract(state.getPoint2(), state.getPoint1(), vector);
+    const center = [0, 0, 0];
+    vtkMath.multiplyAccumulate(state.getPoint1(), vector, 0.5, center);
+    axis.line.source.setCenter(center);
+    const length = vtkMath.normalize(vector);
+    axis.line.source.setDirection(vector);
+    axis.line.source.setHeight(length);
```

Comment:
My understanding of this is that lines will be drawn only between two rotations points ?
But now, lines are longer than this distance.

Reply:
the cylinder will be as long as the previous lines.

Reply:
ok sounds good then!